### PR TITLE
mailsend: Remove more deprecated APIs

### PR DESCRIPTION
**Pull request by neheb**
Fixes compile without them.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

Maintainer: @thess 
Compile tested: mvebu

### CODE REVIEW

**Comment by hnyman** [comment]
@thess @cotequeiroz @neheb 
Any verdict for this PR?

**Comment by neheb** [comment]
This can be tested by disabling deprecated APIs for OpenSSL in make menconfig, which I do in all of my builds.

File: mail/mailsend/patches/0002-Removed-API-deprecated-by-OpenSSL-1.1.0.patch [hunk 1]
```diff
@@ -39,6 +40,36 @@ index d275e6a..51bfb1d 100755
...
+     static const char rnd_seed[]="my huge entropy for rng.. blah";
+     SSL_CTX *ssl_ctx=(SSL_CTX *) NULL;
+     SSL *ssl=NULL;
++#if OPENSSL_VERSION_NUMBER < 0x10100000L
+     SSL_library_init();
+     SSL_load_error_strings();
+-    RAND_seed(rnd_seed,sizeof(rnd_seed));
+     OpenSSL_add_all_algorithms();
++#else
++    OPENSSL_init_ssl(0, NULL);
```
Comment (cotequeiroz) [inline mail/mailsend/patches/0002-Removed-API-deprecated-by-OpenSSL-1.1.0.patch]: I wouldn't make this call, as the default init is done by the library automatically.

**Comment by cotequeiroz** [comment]
I've made a suggestion, but it is fine as is.

**Comment by neheb** [comment]
ping @thess

**Comment by neheb** [comment]
Rebased for CircleCI

**Pull request merged**